File: No_2530/Maximal Score After Applying K Operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<int, vector<int>> maxHeap;
        for(int num : nums) {
            maxHeap.push(num);
        }
        
        long long ret = 0;
        while (k--) {
            int top = maxHeap.top(); maxHeap.pop();
            ret += top;
            maxHeap.push(ceil(top / 3.));
        }
        
        return ret;
    }
};

File: No_2530/Maximal Score After Applying K Operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        max_heap = [-1 * num for num in nums]
        heapq.heapify(max_heap)
        
        ret = 0
        while k != 0:
            num = -1 * heapq.heappop(max_heap)
            ret += num
            heapq.heappush(max_heap, -1 * math.ceil(num / 3))
            k -= 1
            
        return ret